pkg/proxy/proxy.go

```diff
@@ -137,7 +138,7 @@ func (p *Proxy) StartProxy(port string) *http.Server {
 	)
 
 	// routes
-	wg := router.Group("/apis/toolchain.dev.openshift.com/v1alpha1/workspaces")
+	wg := router.Group("/apis/toolchain.dev.openshift.com/v1alpha1/workspaces", p.ensureUserIsNotBanned())
```

Comment:
Instead of adding this middleware to specific routes you can add it to the list of our middleware for all routes:
https://github.com/codeready-toolchain/registration-service/blob/5faca3af920bc27bb3acee382fce5662e1463321/pkg/proxy/proxy.go#L105-L121

And to avoid the situation when it's enforced for unsecured routes (which does not require authentication) you can add the following check: https://github.com/codeready-toolchain/registration-service/blob/5faca3af920bc27bb3acee382fce5662e1463321/pkg/proxy/proxy.go#L390-L392

In this case any new secured route we add in the future it will be covered automatically covered by this middlewere.

Reply:
done in 73039cc4deb732a28efae65e3fe22d677c9494a2